### PR TITLE
feat: transformer refactor

### PR DESCRIPTION
**Pull request by utsabc**
## Description of the change

This PR works towards refactoring transformer and into specific modules and bring MVC architecture for better organization of codebase

- Removed Unnecessary Data files which are not used today

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: services/cdkV2Workflow.service.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    destinationType: string,
+    destHandler: any
+  ): Promise<RouterResponse[]> {
+    const allDestEvents: Object = groupBy(
+      events,
+      (ev: RouterData) => ev.destination?.ID
+    );
+    // TODO: Change the promise type
+    const response: RouterResponse[] = await Promise.all<any>(
+      Object.values(allDestEvents).map(async (destInputArray: RouterData[]) => {
```
Comment (koladilip) [inline services/cdkV2Workflow.service.ts:95]: group by destination id is not required here?

File: services/IWorkFlow.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import {
+  Metadata,
+  ProcessorRequest,
+  ProcessorResponse,
+  RouterData,
+  RouterResponse,
+  TransformationDefaultResponse
+} from "../types/types";
+
+export default interface IWorkFlow {
```
Comment (koladilip) [inline services/IWorkFlow.ts:10]: IWorkflow -> DestinationService (Somehow calling this workflow seems bit odd)

File: services/cdkV2Workflow.service.ts [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+  ProcessorRequest,
+  ProcessorResponse,
+  RouterData,
+  RouterResponse,
+  TransformationDefaultResponse
+} from "../types/types";
+import { generateErrorObject } from "../v0/util";
+import { TRANSFORMER_METRIC } from "../v0/util/constant";
+import IWorkFlow from "./IWorkFlow";
+export default class CDKV2Service implements IWorkFlow {
```
Comment (koladilip) [inline services/cdkV2Workflow.service.ts:15]: CDKV2Service -> CDKV2DestinationService

File: controllers/destination.controller.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+import { ProcessorRequest } from "../types/types";
+import { WorkFlowSelectorService } from "../services/worflowSelector.service";
+import IWorkFlow from "../services/IWorkFlow";
+
+export class DestinationController {
+  public static async destinationVanillaTransformAtProcessor(ctx: Context) {
+    const events = ctx.request.body as ProcessorRequest[];
+    const { version, destination } = ctx.params;
+    const reqParams = ctx.request.query;
+    const destinationHandler = WorkFlowSelectorService.getDestHandler(
```
Comment (koladilip) [inline controllers/destination.controller.ts:11]: destination handler and workflow service both needed, is it for comparison logic?

**Review by koladilip — COMMENTED** [review]
Not full review

File: controllers/destination.controller.ts (hunk 4 above)
Comment (utsabc) [reply controllers/destination.controller.ts:11]: I am passing the handler as an argument for the workflow service. Handler is the core `transform.js` functions which I am decoupling from the workflow logic this way I will be able to handle other intermediate services (comparator or other stuff if required)

File: services/cdkV2Workflow.service.ts (hunk 1 above)
Comment (utsabc) [reply services/cdkV2Workflow.service.ts:95]: Yes I have handled it at line 89

File: services/IWorkFlow.ts (hunk 2 above)
Comment (utsabc) [reply services/IWorkFlow.ts:10]: Shall we create a hierarchy for individual flows or a central hierarchy for all flows (destination flows, source, etc)?

File: services/destination/cdkV1Integration.destination.service.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+            error,
+            errorDTO
+          );
+        }
+      })
+    );
+    return respList;
+  }
+
+  public routerRoutine(
```
Comment (koladilip) [inline services/destination/cdkV1Integration.destination.service.ts:58]: we can have one abstract class with all methods not implemented and each of service implementations can extend and implement only relevant function. for example: CDKV1ServiceDestination only implements processorRoutine

**Review by koladilip — CHANGES_REQUESTED** [review]

**Comment by utsabc** [comment]
Closing the following PR in leu of https://github.com/rudderlabs/rudder-transformer/pull/1710

**Pull request closed without merging**